src/compiler/binder.ts

```diff
@@ -943,7 +943,8 @@ namespace ts {
                 else {
                     return node.kind === SyntaxKind.BinaryExpression && (
                         (<BinaryExpression>node).operatorToken.kind === SyntaxKind.AmpersandAmpersandToken ||
-                        (<BinaryExpression>node).operatorToken.kind === SyntaxKind.BarBarToken);
+                        (<BinaryExpression>node).operatorToken.kind === SyntaxKind.BarBarToken ||
+                        (<BinaryExpression>node).operatorToken.kind === SyntaxKind.QuestionQuestionToken);

```

Comment:
`??` isn't really a logical operator. This results in control-flow treating the branch as something "truthy" not something defined/undefined. 

I'd like to see some tests for control flow.